source/common/network/happy_eyeballs_connection_impl.h

```diff
@@ -73,6 +73,8 @@ class HappyEyeballsConnectionImpl : public ClientConnection,
   bool isHalfCloseEnabled() override;
   std::string nextProtocol() const override;
   // Note, this might change before connect finishes.
+  ConnectionInfoSetter& connectionInfoSetter() override;
+  // Note, this might change before connect finishes.
```

Comment:
@RyanTheOptimist can you do a single pass on this one to make sure it looks correct for happy eyeballs?

Reply:
This seems correct. It's a simple getter, so deferring to the first connection seems reasonable. However, I'm not entirely sure how this method is used. If a caller were to cache the return value and use it later that could be problematic. But in this respect, it's just like connectionInfoProvider().